main.go

```diff
@@ -380,7 +381,17 @@ func (lb *loadBalancer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	handlerFn := func(w http.ResponseWriter, r *http.Request) {
 		proxy.ServeHTTP(w, r)
 	}
-	httpTraceHdrs(handlerFn, w, r, lb.backends[bkIdx])
+
+	cacheHandlerFn := func(w http.ResponseWriter, r *http.Request) {
+		if lb.backends[bkIdx].cacheClient != nil {
+			cacheHandler(handlerFn, w, r, lb.backends[bkIdx])(w, r)
+		} else {
+			handlerFn(w, r)
+		}
+	}
```

Comment:
you can simplify by removing `handlerFn` and do this:
```
385        cacheHandlerFn := func(w http.ResponseWriter, r *http.Request) {                                                                                                                                        
386                if lb.backends[bkIdx].cacheClient != nil {                                                                                                                                                      
387                        cacheHandler(w, r, lb.backends[bkIdx])                                                                                                                                                  
388                } else {                                                                                                                                                                                        
389                        proxy.ServeHTTP(w, r)                                                                                                                                                                   
390                }                                                                                                                                                                                               
391        }                                                                                                                                                                                                       
```

Reply:
actually the handlerFn is required to be passed into the cacheHandler - if the request cannot be cached, it falls back to the handlerFn

Reply:
I have pushed the changes here ... checkout how the code flow can be simplified:
https://github.com/krishnasrinivas/sidekick/commit/9c4d8eb2d4f6b27b83bff4951c3f8e10d2e882c3